dashboard/state_aggregator.py

```diff
@@ -162,24 +233,38 @@ async def list_tasks(self, *, option: ListApiOptions) -> dict:
 
         # Sort to make the output deterministic.
         result.sort(key=lambda entry: entry["task_id"])
-        return {d["task_id"]: d for d in islice(result, option.limit)}
+        return ListApiResponse(
+            result={d["task_id"]: d for d in islice(result, option.limit)},
+            partial_failure_warning=(
```

Comment:
Should the warning be empty if all succeeded? Otherwise how would caller know if it's partial success or full success

Reply:
oh yeah good catch. Idk how I didn't catch this from tests. will fix